src/aave-v2/lens/RatesLens.sol

```diff
@@ -121,8 +128,10 @@ abstract contract RatesLens is UsersLens {
         ) = _getIndexes(_poolToken);
 
         Types.BorrowBalance memory borrowBalance = morpho.borrowBalanceInOf(_poolToken, _user);
+
+        uint256 withdrawnFromPool;
         if (!market.isP2PDisabled) {
-            if (_amount > 0 && delta.p2pSupplyDelta > 0) {
+            if (delta.p2pSupplyDelta > 0) {
```

Comment:
Why remove this check ? 

Also, we can harmonize the if-else statements in this function and in `getNextUserSupplyRatePerYear`. In particular in `getNextUserSupplyRatePerYear` function the statement `_amount > 0 && !market.isP2PDisabled` could be changed into simply `_amount > 0`